echopype/convert/set_groups_ad2cp.py

```diff
@@ -368,4 +368,16 @@ def set_vendor(self) -> xr.Dataset:
         return set_encodings(ds)
 
     def set_sonar(self) -> xr.Dataset:
-        return xr.Dataset()
+        ds = xr.Dataset(
+            attrs={
+                "sonar_manufacturer": "Nortek",
+                "sonar_model": "AD2CP",
+                "sonar_serial_number": "",
+                "sonar_software_name": "",
+                "sonar_software_version": "",
+                "sonar_type": "omnisonar",
```

Comment:
Could you change the following:
- `sonar_type` --> "acoustic Dopper current profiler"?
- `sonar_model` : @imranmaj do you know if there is a place in the file that encodes string like "Signature 1000" ?  I don't seem to remember one on top of my head.

For `sonar_serial_number`, please use the serial number in the parsed data (p.59 of integrator's guide). It is also in the string following "ID" below:

Also, because of how the instrument is set up, I think we should add an attribute `sonar_firmware_version`.  
@imranmaj : This should be the strings following "GETHW". You can find the spec in the guide.
@emiliom : tagging you here to see if you have any suggestions.

The first lines of GETALL that shows the config:
<img width="786" alt="Screen Shot 2021-06-29 at 11 59 32 AM" src="https://user-images.githubusercontent.com/15334215/123853522-65f1a880-d8d2-11eb-9dc9-1187b62d4050.png">

Reply:
> For sonar_serial_number, please use the serial number in the parsed data (p.59 of integrator's guide). It is also in the string following "ID" below:

I believe I did this here: https://github.com/OSOceanAcoustics/echopype/blob/5ea0cc05bdb9f62420b261b6b73fa07652bdf5e9/echopype/convert/set_groups_ad2cp.py#L381-L382

Reply:
> do you know if there is a place in the file that encodes string like "Signature 1000" ? I don't seem to remember one on top of my head.

I don't recall one, but I will look through the data

Reply:
@emiliom the `GETHW` portion is in a key-value format; is it ok to have a dictionary in an attribute?
![image](https://user-images.githubusercontent.com/49664304/123880073-c266bf00-d8f6-11eb-876b-424bad1c5f85.png)

Reply:
>@emiliom the GETHW portion is in a key-value format; is it ok to have a dictionary in an attribute?

I don't think attributes can accommodate dictionaries as such. I recommend transforming it into a string form, like this: "FW:2214, FPGA:185, DIGITAL:I-3 ...". That is, a string in a form that could be easily parsed into a dictionary

Reply:
> `sonar_model` : @imranmaj do you know if there is a place in the file that encodes string like "Signature 1000" ? I don't seem to remember one on top of my head.

I don't know what the string "Signature 1000" represents, given that you're already describing the `sonar_model` as AD2CP. Looking at https://www.nortekgroup.com/products, I see one model that's described as having "AD2CP capabilities", and one model that's specifically called "Signature1000" (which matches this model descriptor in the NERC Vocabulary Server: http://vocab.nerc.ac.uk/collection/L22/current/TOOL1009/)

Reply:
Signature 1000 is the specific instrument model that is used to collect the data. "Signature" is a family/series of instruments Nortek developed that have the additional "echosounder mode" in addition to the common velocity sampling capability. It seems that it is being used interchangeably with "AD2CP", with `.ad2cp` being the file extension for all instruments in this family.